### PR TITLE
Feature/firedoc google docs api control

### PR DESCRIPTION
**Pull request by wafer-bw**
https://drive.google.com/file/d/1zAIxt6jStFvOxG8i1T1MTyWNQp9YHewF/preview

The title of the fire can now be passed `/firetest DC Outage`

Nebo now automatically creates a Fire Doc using Google Docs API  and updates the content to include the correct title and a first bullet point indicating when the fire was called.

The link to the doc is provided in the slack message from Nebo

Also, you'll notice we moved index.go into a folder named api so that this repo now follows vercel's [serverless function](https://vercel.com/docs/serverless-functions/introduction) conventions

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,10 @@
...
+
 .now
-.env
\ No newline at end of file
+.vercel
+
+.idea
+.vscode
+
+package-lock.json
```
Comment (codeallthethingz) [inline .gitignore:9]: you must commit package-lock.json always google whether something should be in a repo or not.  The internet usually has an opinion.

File: api/index.go [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+	if v, ok := os.LookupEnv(key); !ok {
+		panic(fmt.Sprintf("Variable %s is not set", key))
+	} else if v == "" {
+		panic(fmt.Sprintf("Variable %s is blank", key))
+	} else {
+		return v
+	}
+}
+
+func getEnvironmentValues() (string, string, string, string, string, string, string, string, string) {
```
Comment (codeallthethingz) [inline api/index.go:332]: why didn't you use the env library?

File: api/index.go [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+		return
+	}
+
+	if !s.ValidateToken(slackVerificationCode) {
+		w.WriteHeader(http.StatusUnauthorized)
+		w.Write([]byte("slack verification failed"))
+		return
+	}
+
+	salesForceVars := []string{sfURL, sfUser, sfPassword, sfToken}
```
Comment (codeallthethingz) [inline api/index.go:52]: you've lost which environment variable was causing the problem, that's important for people to see what's going wrong.

File: api/index.go [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+	}
+
+	gapiVars := []string{gapiEmail, gapiPrivateKey, fireDocFolderID}
+	if gapiDAO == nil && !containsEmptyString(gapiVars...) {
+		gapiDAO = gapi.NewDAO(gapiEmail, gapiPrivateKey, fireDocFolderID)
+	}
+
+	w.Header().Set("Content-type", "application/json")
+	switch s.Command {
+	case "/rep", "/alpha-nebo", "/nebo":
```
Comment (codeallthethingz) [inline api/index.go:74]: the move to api should be a separate pull request as I can no longer see what the actual changes are.

**Review by codeallthethingz — CHANGES_REQUESTED** [review]
aborted the code review as the api move makes all lines changed.  Do the api move as a separate PR after this one.

**Comment by wafer-bw** [comment]
Cancelling and splitting

**Pull request closed without merging**